### ISSUE TITLE
Rule Proposal: prefer-nullish-coalescing

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#471]
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator

https://github.com/tc39/proposal-nullish-coalescing

```js
let foo = (undefined || null) || bar;

// equals

let foo = (undefined || null) ?? bar;
```

anyway, it not safe for `foo || bar` -> `foo ?? bar`, we should [use ESLint suggestion api](#441)

**Comment by sindresorhus** [comment]
For inspiration: https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/prefer-nullish-coalescing.md

This obviously has to wait until ESLint supports `??`.

**Comment by EvgenyOrekhov** [comment]
Support for nullish coalescing operator was added in [ESLint 7.2.0](https://eslint.org/blog/2020/06/eslint-v7.2.0-released).

**Comment by sindresorhus** [comment]
This is now accepted. PR welcome.

**Comment by sindresorhus** [comment]
If anyone wants to work on this, see the initial attempt and feedback in https://github.com/sindresorhus/eslint-plugin-unicorn/pull/1388.

**Comment by geoffswift** [comment]
I appreciate there is some difficulty with the feasability for supporting `||` but could we at least get ternaries simplified e.g.

```javascript
// bad
example === undefined ? 'wow' : example;

// good
example ?? 'wow'
```

And likewise
```javascript
// bad
example !== undefined ? example : 'wow';

// good
example ?? 'wow'
```
